### ISSUE TITLE
IDN E-mail with username in Unicode is not accepted

### ISSUE THREAD

**Issue by rosmanov** [PHPMailer/PHPMailer#668]
Validation fails, if the E-mail username contains non-ASCII characters:

```php
<?php
$m = new PHPMailer();
$m->CharSet = 'utf-8';
$m->ContentType = 'text/html';
$m->Encoding = 'base64';

$e = 'османов@письмо.рф';
$ep = $m->punyencodeAddress($e);
var_dump($ep);
var_dump($m->validateAddress($ep, 'pcre8'));
```
Actual output:
```
string(37) "османов@xn--h1aigbl0e.xn--p1ai"
bool(false)
```
Expected output:
```
string(37) "османов@xn--h1aigbl0e.xn--p1ai"
bool(true)
```

**Comment by Synchro** [comment]
This is intentional, and mentioned in #88. UTF-8 in the local part of email adresses is only supported via [RFC6531](https://tools.ietf.org/html/rfc6531#section-3.3), and that has very little adoption as yet (pretty much only gmail). What's more of a problem is that if you use such an address and then try to send it via a server which doesn't support it, you're stuck with having to do everything in [RFC6857](https://tools.ietf.org/html/rfc6857), which really isn't pretty and requires architectural changes (because a message cannot be rendered until *after* you have connected to the server). As part of that process, punycode *can* be applied to the local part of addresses via [RFC5891](https://tools.ietf.org/html/rfc5891) and [RFC5890](https://tools.ietf.org/html/rfc5890), but support for that is even worse than for RFC6531.

In short, doing this is an enormous amount of work, so if you feel like writing it and doing a PR, be my guest.

**Comment by UA2018** [comment]
Hi,

You may find the following information useful in developing systems, internet applications that accepts IDNs, EAI and New gTLDs. 

Universal Acceptance Quick Guide:
https://uasg.tech/wp-content/uploads/documents/UASG005-en-digital.pdf

Good Practice Recommendations:
https://uasg.tech/wp-content/uploads/documents/UASG007-en-digital.pdf
